subsys/bluetooth/controller/ll_sw/ull_filter.c

```diff
@@ -1103,12 +1103,13 @@ static void conn_rpa_update(uint8_t rl_idx)
 	for (handle = 0U; handle < CONFIG_BT_MAX_CONN; handle++) {
 		struct ll_conn *conn = ll_connected_get(handle);
 
-		/* The RPA of the connection matches the RPA that was just resolved */
-		if (conn &&
-		    conn->peer_addr_type < 2U &&
-		    !memcmp(conn->peer_addr, rl[rl_idx].curr_rpa.val, BDADDR_SIZE)) {
-			memcpy(conn->peer_addr, rl[rl_idx].id_addr.val, BDADDR_SIZE);
-			conn->peer_addr_type += 2U;
```

Comment:
@wopu-ot What is the reason to store Public or Random ID used for resolvable private address? 1 bit device address type of public or random should suffice for same peer checks, right?